src/cobra/core/reaction.py

```diff
@@ -1,44 +1,52 @@
-# -*- coding: utf-8 -*-
-
 """Define the Reaction class."""
 
-from __future__ import absolute_import, print_function
-
 import hashlib
 import re
 from collections import defaultdict
 from copy import copy, deepcopy
 from functools import partial
 from math import isinf
 from operator import attrgetter
-from typing import FrozenSet
+from typing import (
+    TYPE_CHECKING,
+    AnyStr,
+    Dict,
+    FrozenSet,
+    Iterable,
+    Iterator,
+    List,
+    Optional,
+    Sequence,
+    Set,
+    Tuple,
+    Union,
+)
 from warnings import warn
 
-from future.utils import raise_from, raise_with_traceback
 
-from cobra.core.configuration import Configuration
-from cobra.core.gene import GPR, Gene
-from cobra.core.metabolite import Metabolite
-from cobra.core.object import Object
-from cobra.exceptions import OptimizationError
-from cobra.manipulation import remove_genes
-from cobra.util.context import get_context, resettable
-from cobra.util.solver import (
+if TYPE_CHECKING:
+    from optlang.interface import Variable
+    from . import Model, Solution
```

Comment:
Apologies for not pointing out earlier: for imports under `TYPE_CHECKING`, you will need the full import path like so: `from cobra import Model, Solution`.

Reply:
So, static type checkers like `mypy` need to import `cobra` just like `pytest` does for testing. `TYPE_CHECKING` is respected by static type checkers. From my experience of using `mypy`, it reports error. Since, we don't have a static type checking in CI as of now, it will pass the checks. It will be good to have it for future.

Reply:
Done